ctsit/nacculator
# Feature/tip module

This change introduces the Telephone Initial Packet module to NACCulator (a counterpart to the Telephone Followup Packet, which captures regular UDS visit data alongside in-person initial and followup visits). It has the same functionality as every other module in NACCulator's code.

This change was made to add more of NACC's options to NACCulator processing. Initial visits are important for establishing participants in NACC's database, and if centers are unable to submit their TIP data, then they will be unable to submit any subsequent data either.


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. From the TESTING 1Florida Alzheimer's Disease Research Center project in REDCap, download the available sample data.
2. Make sure you have nacculator_cfg.ini.example set up into nacculator_cfg.ini according to the README.
3. From the command-line, run:
$ redcap2nacc -f fixHeaders -meta nacculator_cfg.ini <path/to/sample_data_input.csv >filtered_output.csv
$ redcap2nacc -tip <filtered_output.csv >test_output.txt
4. The output should look like this: 
output.txt will have one participant with clear, standard NACC-formatted data that is able to be submitted through their portal. The error printout will return one error due to a caught blanking rule violation.
5. The output should currently only have these two PTIDs; everything else in the testing project is non-telephone initial packet data.

- Verify the thing does this: Processes Telephone Initial Packet visits while checking for errors and warnings due to blanking rules, field lengths, incorrect characters in Char fields etc
- Verify the thing does not do that: This change does not affect NACCulator's ability to process any other packet types.


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
